### PR TITLE
WIP: Add org.srb2.SRB2Kart

### PR DESCRIPTION
**Pull request by cbm755**
This is a work-in-progress.  I'm not a developer of this game but I plan to [submit it upstream](https://git.magicalgirl.moe/KartKrew/Kart-Public/issues/5)

I'm posting here early so I can get the bot to test some other arches!  But of course, feedback welcome, either here or at https://gitlab.com/cbm755/org-srb2-SRB2Kart

### CODE REVIEW

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

File: org.srb2.SRB2Kart.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                        "cd tmp; 7za x ../srb2kart-v12-Installer.exe",
+                        "mkdir -p assets/installer/",
+                        "cp tmp/srb2.srb assets/installer/",
+                        "cp tmp/{patch,gfx,textures,chars,maps}.kart assets/installer/",
+                        "cp tmp/{music,sounds,bonuschars}.kart assets/installer/"
+                    ]
+                },
+                {
+                    "type": "file",
+                    "path": "org.srb2.SRB2Kart.appdata.xml"
```
Comment (LeandroStanger) [inline org.srb2.SRB2Kart.json:90]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

**Comment by cbm755** [comment]
bot, build org.srb2.SRB2Kart

**Comment by ghisvail** [comment]
My bad, I submitted #1526 without checking the list of open PRs first. I am the maintainer of the flatpak for [SRB2](https://github.com/flathub/org.srb2.SRB2) from which SRB2Kart is derived from.

I'd be happy to co-maintain the flatpak for SRB2Kart with you if you feel like it.

File: org.srb2.SRB2Kart.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: FSFAP
```
Comment (bilelmoussaoui) [inline org.srb2.SRB2Kart.json:1]: I would avoid having a comment in the json file :) you can add a license to the repository if you want to

**Comment by cbm755** [comment]
@ghisvail that would be great!  I learned a lot just glancing at your manifest!  But haven't looked carefully yet...

What do you suggest going forward?  Should I compare and submit a merge request if there is anything useful in mine?  Or do you want to?

**Comment by ghisvail** [comment]
@cbm755 I initially tried the CMake approach with SRB2(Kart) and found the build required too much patching to be maintainable long term. Besides, upstream still uses the Makefiles for their [very own PPA](https://launchpad.net/~kartkrew/+archive/ubuntu/srb2kart), which is a sign.

As such, I would propose to use my PR as a basis and improve on it together. Something @barthalion rightfully suggested is to use Flatpak's [extra data](https://github.com/flathub/org.srb2.SRB2/issues/1) to provide the proprietary assets at install time.

It would be great if you could have a look at it for `SRB2Kart` whilst I deal with it for `SRB2`. How does it sound?

**Comment by cbm755** [comment]
Sounds good!   And +1 for `extra data`: I was going to try that too.

Are you an upstream developer on these projects?  I gather we need upstream onboard (or at least not opposed) to get these into Flathub.  And it would be good to get the metainfo.xml and .desktop into upstream...

**Comment by ghisvail** [comment]
> Are you an upstream developer on these projects?

I am just an occasional player of both games.

>  I gather we need upstream onboard.

I pinged them on their Discord channel, and filed an [issue](https://github.com/STJr/SRB2/issues/459) on GitHub. No reply yet.

**Comment by cbm755** [comment]
Me too: https://git.magicalgirl.moe/KartKrew/Kart-Public/issues/5

**Comment by ghisvail** [comment]
> And it would be good to get the metainfo.xml and .desktop into upstream...

Baby steps :+1:

File: org.srb2.SRB2Kart.json (hunk 2 above)
Comment (cbm755) [reply org.srb2.SRB2Kart.json:1]: ok.  Is there a validator tool?  `flatpak-builder` didn't complain...

But probably I will just switch to yaml, which I think explicitly allows comments.

**Comment by cbm755** [comment]
Closing in favour of #1526

**Pull request closed without merging**